src/nvim/eval.c

```diff
@@ -6816,20 +6816,20 @@ char *set_cmdarg(exarg_T *eap, char *oldarg)
   }
 
   if (eap->force_ff != 0) {
-    snprintf(newval + strlen(newval), newval_len, " ++ff=%s",
+    snprintf(newval + strlen(newval), newval_len - strlen(newval), " ++ff=%s",
```

Comment:
Do we need `assert(newval_len - strlen(newval) > 0)` before each of these?

Would it be worth capturing the return value of `snprintf` from the previous calls, to avoid the redundant strlen()?

Does the call on line 6837 also need this treatment?

Reply:
Line 6837 doesn't append but overwrites, should it append? The STRCPY also looks bad.

Reply:
I've uploaded a complete overhaul of the function. This should work.